### DISCUSSION TITLE
Revalidate is re-rendering root layout when I believe it should not

### DISCUSSION THREAD

**Discussion by mattkauffman23** [vercel/next.js#59037] [Help]
### Summary

It's possible I don't understand how revalidatePath is supposed to work, but wanted to confirm.

I have a server action that revalidates a path like this:
`revalidatePath("/(dashboard)/connections", "page");`

I expected that to trigger a re-render for any pages / layouts downstream of `connections` but I'm seeing the root layout re-render. It's particularly noticeable because I'm using custom fonts that cause a bunch of layout shifts when the root layout reloads them.

My question is does revalidating a route re-render the entire page for that route inclusive of ancestor layouts or should it only re-render pages / layouts from that route segment down?

**Comment by festusyuma** [comment]
Hello @mattkauffman23 . did you find any solution to this. I am currently experiencing the same problem

**Comment by mattkauffman23** [comment]
Unfortunately I did not.

**Reply by festusyuma** [reply]
Alright, thank.

**Reply by GuysmoB** [reply]
Same question for me :
In my example, console.log('children'); is printed when I click on the test button that triggers the revalidate function.
We agree it's not supposed to happen, right?

```
'use client';

import { revalidatePathsFn } from '#/lib/actions';
import React, { useEffect } from 'react';

export function ClickCounter() { 

useEffect (() => {
  console.log('ClickCounter');
} , []);

  return (
    <>
    <button
      onClick={() => revalidatePathsFn(['/'])}
      className="whitespace-nowrap rounded-lg bg-gray-700 px-3 py-1 text-sm font-medium tabular-nums text-gray-100 hover:bg-gray-500 hover:text-white"
      >
      TEST 
    </button>
      </>
  );
}

```

```
'use client';

import React, { FC, useEffect } from 'react';

interface ITest {
	children: React.ReactNode;
}

const Test: FC<ITest> = ({ 	children }) => {

useEffect (() => {
  console.log('children');
} , [children]);

  return (
    <>
    {children}
    </>
  );
}

export default Test;

```

```
...
<div className="self-start">
        <Test><ClickCounter /></Test>
          
        </div>
```

**Comment by a89529294** [comment]
Same issue here, I tried calling revalidatePath with different arguments, but the console log in the root layout always gets printed on the server side.

**Comment by JayMGurav** [comment]
any updates on this i took face the same issue. `revalidatePath('/', 'layout')` doesn't re-render the root layout
